### PR TITLE
First pass at a dark theme

### PR DESCRIPTION
**Pull request by goldbattle**
I was unable to get the full esphome stack running on my windows machine, but was able to host a small local web server with the static files in the [web.esphome.io](https://github.com/esphome/dashboard/tree/main/web.esphome.io) folder. This uses the user's `prefers-color-scheme: dark` setting to change some css values so it is a bit better on the eyes. This should close https://github.com/esphome/dashboard/issues/347.

Here are some screenshots:

screencapture-127-0-0-1-8080-2023-02-05-03_11_45

screencapture-127-0-0-1-8080-2023-02-05-03_12_16

I was thus unable to test any screens past these ones, so the editor will still be the light theme.
I believe this would require changing the following code.
Maybe this should be a config setting, or maybe [check with js](https://stackoverflow.com/a/57795495/7718197) and toggle to `vs-dark` with a darker `editor.foreground`?
https://github.com/esphome/dashboard/blob/e60c7d936de424b9360ab892317723c4277404af/src/editor/monaco-provider.ts#L57-L65

### CODE REVIEW

**Comment by Misiu** [comment]
@goldbattle I have some progress on the dark theme you might take a look: https://github.com/Misiu/esphome_dashboard/commit/6c8b82afd146e42280add18ab008cd17643ab4b6
Sadly not enough time to move this forward in the near future.

**Comment by goldbattle** [comment]
Thanks for the pointer. I tried to change as little as possible here in this PR.

I am using esphome in home assistant, and I couldn't find that css file.
Is the `esphome-2.css` is shared between this and the website?

**Comment by balloob** [comment]
This theme does not look good. I would suggest that you take the dark themes of Home Assistant and use those color values.

**Comment by goldbattle** [comment]
Thanks, here are the images of what it looks like using the CSS `#111111` background, `#1c1c1c` card, and `#e1e1e1` text from the [frontend](https://github.com/home-assistant/frontend/blob/d15d339782b89dab0d6d9c029abafe5c140c0d63/src/resources/styles.ts#L3-L7) repo. Let me know any additional comments.

Page 1:
image
image


Page 2:
image
image

File: web.esphome.io/src/dashboard/unsupported-card.ts [hunk 1]
```diff
@@ -23,6 +23,9 @@ class EWUnsupportedCard extends LitElement {
       esphome-card {
         --status-color: var(--alert-error-color);
       }
+      .card-actions mwc-button {
+        --mdc-theme-primary: var(--card-text-color);
```
Comment (balloob) [inline web.esphome.io/src/dashboard/unsupported-card.ts:27]: It looks wrong to set a theme color to a text color variable.

Comment (goldbattle) [reply web.esphome.io/src/dashboard/unsupported-card.ts:27]: This is what is currently done to change the color:
https://github.com/esphome/dashboard/blob/270f7e462bcf9e016f1169bca40548e091586ae8/web.esphome.io/src/dashboard/esp-connect-card.ts#L81-L83

In the docs it says "Color of a flat or outlined button. Background color of a raised button."
https://www.npmjs.com/package/@material/mwc-button#user-content-global-custom-properties

**Pull request closed without merging**